### PR TITLE
Abstract getSubgraphAroundNode

### LINKED ISSUE #214: Mini cleanup on solvable for GetSubgraphAroundNode on LightGraphs
Hi @Affie,

I added this https://github.com/JuliaRobotics/DistributedFactorGraphs.jl/blob/master/src/LightDFG/services/LightDFG.jl#L395 - but I think it will behave differently than the other drivers because it doesn't grow from 1 to `distance` neighbors. E.g. if in a graph x0 is solvable, x0x1f1 is not solvable, and x1 is solvable, asking for a subgraph around x0 distance=2,solvable=1 will return x1 even though x0x1f1 isn't solvable. 

I'm checking and adding a test for it, but do you mind taking a look at whether we can change it if needed?

### LINKED ISSUE #261: [BUG] subgraph functions not working for LightDFG
```julia
julia> sfg = buildCliqSubgraph(fg, tree, :x1)
ERROR: getSubgraphAroundNode not implemented for LightDFG{SolverParams,DFGVariable,DFGFactor}
Stacktrace:
 [1] error(::String) at ./error.jl:33
 [2] #getSubgraphAroundNode#19(::Int64, ::typeof(getSubgraphAroundNode), ::LightDFG{SolverParams,DFGVariable,DFGFactor}, ::DFGVariable, ::Int64, ::Bool, ::GraphsDFG{SolverParams}) at /home/dehann/.julia/dev/DistributedFactorGraphs/src/services/AbstractDFG.jl:430
 [3] (::getfield(DistributedFactorGraphs, Symbol("#kw##getSubgraphAroundNode")))(::NamedTuple{(:solvable,),Tuple{Int64}}, ::typeof(getSubgraphAroundNode), ::LightDFG{SolverParams,DFGVariable,DFGFactor}, ::DFGVariable, ::Int64, ::Bool, ::GraphsDFG{SolverParams}) at ./none:0
 [4] #buildSubgraphFromLabels!#192(::GraphsDFG{SolverParams}, ::Int64, ::Array{Symbol,1}, ::typeof(buildSubgraphFromLabels!), ::LightDFG{SolverParams,DFGVariable,DFGFactor}, ::Array{Symbol,1}) at /home/dehann/.julia/dev/DistributedFactorGraphs/src/needsahome.jl:34
 [5] (::getfield(DistributedFactorGraphs, Symbol("#kw##buildSubgraphFromLabels!")))(::NamedTuple{(:subfg, :solvable, :allowedFactors),Tuple{GraphsDFG{SolverParams},Int64,Array{Symbol,1}}}, ::typeof(buildSubgraphFromLabels!), ::LightDFG{SolverParams,DFGVariable,DFGFactor}, ::Array{Symbol,1}) at ./none:0
 [6] buildCliqSubgraph(::LightDFG{SolverParams,DFGVariable,DFGFactor}, ::BayesTree, ::IncrementalInference.TreeClique, ::GraphsDFG{SolverParams}) at /home/dehann/.julia/dev/IncrementalInference/src/CliqStateMachineUtils.jl:1045
 [7] buildCliqSubgraph(::LightDFG{SolverParams,DFGVariable,DFGFactor}, ::BayesTree, ::Symbol, ::GraphsDFG{SolverParams}) at /home/dehann/.julia/dev/IncrementalInference/src/CliqStateMachineUtils.jl:1064 (repeats 2 times)
 [8] top-level scope at REPL[7]:1
```

### PR DESCRIPTION
**Pull request by Affie**
Close #214
Close #261

### CODE REVIEW

**Comment by dehann** [comment]
awesome thanks!

File: src/needsahome.jl [hunk 1]
```diff
@@ -25,7 +25,7 @@ function buildSubgraphFromLabels!(dfg::G,
                                   syms::Vector{Symbol};
                                   subfg::AbstractDFG=(G <: InMemoryDFGTypes ? G : GraphsDFG)(params=getSolverParams(dfg)),
```
Comment (Affie) [inline src/needsahome.jl]: Note GraphsDFG is hard coded here. Depending on the IIF use case, you might have to change it here if the default is changed in IIF.

Comment (dehann) [reply src/needsahome.jl]: Yeah, think you defined a more standard `InMemDFGType` -- can we just use that?

Comment (dehann) [reply src/needsahome.jl]: see #266

**Comment by dehann** [comment]
This is great, thanks for jumping on and fixing!

**Comment by dehann** [comment]
think this replaced #262

**Pull request merged**